src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzuerArtifacts.tsx

```diff
@@ -0,0 +1,389 @@
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useMemo } from 'react'
+import {
+  Formik,
+  Layout,
+  Button,
+  StepProps,
+  Text,
+  ButtonVariation,
+  SelectOption,
+  FormInput,
+  FormikForm,
+  MultiTypeInputType
+} from '@harness/uicore'
+import * as Yup from 'yup'
+import { FontVariation } from '@harness/design-system'
+import { defaultTo, memoize } from 'lodash-es'
+import { useParams } from 'react-router-dom'
+import type { IItemRendererProps } from '@blueprintjs/select'
+import type { FormikProps } from 'formik'
+import { useStrings } from 'framework/strings'
+import type { GitQueryParams, ProjectPathProps } from '@common/interfaces/RouteInterfaces'
+import { useQueryParams } from '@common/hooks'
+
+import {
+  ConnectorConfigDTO,
+  useListProjectsForAzureArtifacts,
+  useListFeedsForAzureArtifacts,
+  useListPackagesForAzureArtifacts,
+  AzureDevopsProject,
+  AzureArtifactsFeed,
+  AzureArtifactsPackage
+} from 'services/cd-ng'
+import { getConnectorIdValue, isFieldFixedAndNonEmpty } from '@pipeline/components/ArtifactsSelection/ArtifactUtils'
+import { getGenuineValue } from '@pipeline/components/PipelineSteps/Steps/JiraApproval/helper'
+import { RepositoryFormatTypes } from '@pipeline/utils/stageHelpers'
+import { EXPRESSION_STRING } from '@pipeline/utils/constants'
+import type { AzureArtifactsRegistrySpec } from 'services/pipeline-ng'
+import ItemRendererWithMenuItem from '@common/components/ItemRenderer/ItemRendererWithMenuItem'
+import { scopeOptions } from '@pipeline/components/ArtifactsSelection/ArtifactHelper'
+import { NoTagResults } from '@pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/ArtifactImagePathTagView/ArtifactImagePathTagView'
+import type { ImagePathProps } from '../../../ArtifactInterface'
+import css from '../../ArtifactConnector.module.scss'
+
+export const packageTypeOptions: SelectOption[] = [
+  { label: 'Maven', value: RepositoryFormatTypes.Maven },
+  { label: 'NuGet', value: RepositoryFormatTypes.NuGet }
+]
+
+function FormComponent(
+  props: StepProps<ConnectorConfigDTO> &
+    ImagePathProps<AzureArtifactsRegistrySpec> & { formik: FormikProps<AzureArtifactsRegistrySpec> }
+): React.ReactElement {
+  const { prevStepData, previousStep, formik } = props
+  const { getString } = useStrings()
+  const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
+  const { repoIdentifier, branch } = useQueryParams<GitQueryParams>()
+  const commonParams = {
+    accountIdentifier: accountId,
+    projectIdentifier,
+    orgIdentifier,
+    repoIdentifier,
+    branch
+  }
+
+  const connectorRefValue = defaultTo(getGenuineValue(prevStepData?.connectorId?.value), '')
```

Comment:
```suggestion
  const connectorRef = getConnectorIdValue(prevStepData)
```
After this updated update use of connectorRef